src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts

```diff
@@ -0,0 +1,284 @@
+        await action.click();
+        expect(actionSpy).toHaveBeenCalledWith(i);
+      }
+    });
+
+    it('should expose secondary action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'secondaryActionClicked');
+      const actions = summaryActionBarFixture.secondaryActions;
+
+      for (let i = 0; i < actions.length; i++) {
+        const action = actions[i];
+
+        // verify button state
+        expect(action.buttonText).toEqual(testComponent.secondaryActions[i].buttonText);
+        expect(action.isDisabled).toEqual(testComponent.secondaryActions[i].isDisabled);
+
+        // verify the click methods work on the button state
+        await action.click();
+        expect(actionSpy).toHaveBeenCalledWith(i);
+      }
+    });
+
+    it('should expose secondary action button properties', async () => {
+      // modify the defaults
+      testComponent.secondaryActions = [
+        {
+          buttonText: 'some action',
+          isDisabled: true,
+          click: () => testComponent.secondaryActionClicked(0)
+        }
+      ];
+      fixture.detectChanges();
+      await fixture.whenStable();
+
+      const actions = summaryActionBarFixture.secondaryActions;
+
+      for (let i = 0; i < actions.length; i++) {
+        const action = actions[i];
+
+        // verify button state
+        expect(action.buttonText).toEqual(testComponent.secondaryActions[i].buttonText);
+        expect(action.isDisabled).toEqual(testComponent.secondaryActions[i].isDisabled);
+      }
+    });
+
+  });
+
+  describe('cancel button', () => {
+
+    it('should expose cancel action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
+      const action = summaryActionBarFixture.cancelAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
+      const action = summaryActionBarFixture.cancelAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button properties', async () => {
```

Comment:
Could we say `should expose cancel action button properties if the properties change?` That seems to be what we are testing. I just want to be clear when looking down the road about what we are testing compared to the firs test.